### PR TITLE
Migration doc

### PR DESCRIPTION
**Pull request by admiyo**
Wanted to get this officially on as a pulle request.  Going to pull out the API docs parts.

### CODE REVIEW

File: docs/live-migration.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+ associated Pod that contains the VM as a process. When a Pod is
+ scheduled onto a node, it stays there until it is deleted. and
+ initially has the same constraint. A Pod is completely immutable
+ after creation. A VM is mutable regarding to scheduling (cluster)
+ after creation, but those changes will not take effect until a
+ migration takes place.  For example, to pin a VM to a specific node,
+ the VM might be launched with the node selection criteria
+
+```
+nodeSelector:
```
Comment (stu-gott) [inline docs/live-migration.md]: trailing whitespace

File: docs/live-migration.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+A migration also has an associated pod that runs the migration
+process. When the migration completes, the process running in the
+pod's container exits.  The return code of the process indicates the
+status of the migration.  This is translated into the status field of
+the migration object.
+
+Because the Kubernetes design calls for long polling the API server,
+identifying changes, and asynchronous communication between
+components, the flow is somewhat complex and spans multiple processes,
+containers, and nodes.
```
Comment (stu-gott) [inline docs/live-migration.md]: trailing whitespace

File: docs/live-migration.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+containers, and nodes. 
+
+Participants:
+
+Migration -- The Third Party Resource that represents a  migration
+  Request created by User.
+
+MigrationController -- Controller watching for migration
+  requests that creates the VM target Pod
+
```
Comment (stu-gott) [inline docs/live-migration.md]: whitespace

File: docs/live-migration.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    name: testvm
+```
+
+The selector section indicates which vm objects are to be
+migrated. The name field shown here would match a vm with the `name`
+value of `testvm.`
+
+
+To list all migrations:
+
```
Comment (stu-gott) [inline docs/live-migration.md]: Number of lines between text sections is inconsistent. Should it be 1 or 2?

File: docs/live-migration.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+spec:
+  selector:
+    name: testvm
+status:
+  phase: Succeeded
+
+```
+
+Conditions can contain the following states: True, False and Unknown
+with additional metadata like timestamps. See e.g.
```
Comment (stu-gott) [inline docs/live-migration.md]: "See e.g." references a missing example?

File: docs/live-migration.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+To cancel:
+
+`kubectl delete migrations <migration_name>`
+
+
+ A Virtual machine is implemented as a Third Party Resource
+ (TPR). Each successfully running virtual machine object has an
+ associated Pod that contains the VM as a process. When a Pod is
+ scheduled onto a node, it stays there until it is deleted. and
```
Comment (stu-gott) [inline docs/live-migration.md]: nit: "and" should be "Also" (capitalized).

File: docs/live-migration.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+ the VM might be launched with the node selection criteria
+
+```
+nodeSelector: 
+  kubernetes.io/hostname: node0
+```
+
+Which would pin it to node0.  In order to migrate to another node, the
+user would first have to remove (via put or patch) the key and value
+`kubernetes.io/hostname: node0`.  This *would not* force a
```
Comment (stu-gott) [inline docs/live-migration.md]: Asterisks in markdown is italics. I think delimiting with underscores makes it bold. Is that what you wanted?

File: docs/live-migration.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+migration, it would only  *allow* a migration to take place.
+
+A migration also has an associated pod that runs the migration
+process. When the migration completes, the process running in the
+pod's container exits.  The return code of the process indicates the
```
Comment (stu-gott) [inline docs/live-migration.md]: Inconsistent spacing between sentences. In most of the document a single space is used between sentences. In this paragraph 2 are used.

File: docs/live-migration.md [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+the migration object.
+
+Because the Kubernetes design calls for long polling the API server,
+identifying changes, and asynchronous communication between
+components, the flow is somewhat complex and spans multiple processes,
+containers, and nodes. 
+
+Participants:
+
+Migration -- The Third Party Resource that represents a  migration
```
Comment (stu-gott) [inline docs/live-migration.md]: two spaces in middle of a sentence. "  migration"

File: docs/live-migration.md [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+  Request created by User.
+
+MigrationController -- Controller watching for migration
+  requests that creates the VM target Pod
+  
+MigrationPodController -- Controller that Watches for state changes on
```
Comment (stu-gott) [inline docs/live-migration.md]: "Watches" should not be capitalized

File: docs/live-migration.md [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+
+MigrationPod -- Pod which contains the process that runs the migration
+
+VMSourcePod -- Pod which contains the virtual machine prior to the
+  migration
+
+VMDestinationPod -- Pod which contains the virtual machine after the
+  migration
+
+PodController  -- Controller's primary job is managing the Pods for
```
Comment (stu-gott) [inline docs/live-migration.md]: extra space after "PodController"

Comment (admiyo) [inline docs/live-migration.md]: Done

File: docs/live-migration.md (hunk 10 above)
Comment (admiyo) [inline docs/live-migration.md]: Done

File: docs/live-migration.md (hunk 3 above)
Comment (admiyo) [inline docs/live-migration.md]: Done

File: docs/live-migration.md (hunk 9 above)
Comment (admiyo) [inline docs/live-migration.md]: Done

File: docs/live-migration.md (hunk 2 above)
Comment (admiyo) [inline docs/live-migration.md]: Done

File: docs/live-migration.md (hunk 8 above)
Comment (admiyo) [inline docs/live-migration.md]: Done

File: docs/live-migration.md (hunk 7 above)
Comment (admiyo) [inline docs/live-migration.md]: Italics is appropriate here.

File: docs/live-migration.md (hunk 1 above)
Comment (admiyo) [inline docs/live-migration.md]: Done

File: docs/live-migration.md (hunk 6 above)
Comment (admiyo) [inline docs/live-migration.md]: Dropped that sentence, as it does not make sense.

File: docs/live-migration.md (hunk 5 above)
Comment (admiyo) [inline docs/live-migration.md]: Dropped. Don't have an example.

File: docs/live-migration.md (hunk 4 above)
Comment (admiyo) [inline docs/live-migration.md]: 1

File: docs/live-migration.md [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+  uid: 230cb349-0972-11e7-b9cf-525400b9ab10
+spec:
+  selector:
+    name: testvm
+status:
+  phase: Succeeded
+
+```
+
+Conditions can contain the following states: True, False and Unknown
```
Comment (rmohr) [inline docs/live-migration.md]: Could you explain what you mean by `conditions`?

File: docs/live-migration.md [hunk 13]
```diff
@@ -0,0 +1,125 @@
...
+ scheduled onto a node, it stays there until it is deleted. A Pod is
+ completely immutable after creation. A VM is mutable regarding to
+ scheduling (cluster) after creation, but those changes will not take
+ effect until a migration takes place.  For example, to pin a VM to a
+ specific node, the VM might be launched with the node selection
+ criteria.
+ 
+
+```
+nodeSelector:
```
Comment (rmohr) [inline docs/live-migration.md]: yaml

File: docs/live-migration.md [hunk 14]
```diff
@@ -0,0 +1,125 @@
...
+migrated. The name field shown here would match a vm with the `name`
+value of `testvm.`
+
+To list all migrations:
+
+`kubectl get migrations`
+
+To query a specific migration:
+
+`kubectl get migrations <migration_name>`
```
Comment (stu-gott) [inline docs/live-migration.md]: I think the user would need to specify `kubectl get migrations <migration_name> -o yaml` to get the output shown.

File: docs/live-migration.md [hunk 15]
```diff
@@ -0,0 +1,125 @@
...
+ A Virtual machine is implemented as a Third Party Resource
+ (TPR). Each successfully running virtual machine object has an
+ associated Pod that contains the VM as a process. When a Pod is
+ scheduled onto a node, it stays there until it is deleted. A Pod is
+ completely immutable after creation. A VM is mutable regarding to
+ scheduling (cluster) after creation, but those changes will not take
+ effect until a migration takes place.  For example, to pin a VM to a
+ specific node, the VM might be launched with the node selection
+ criteria.
+
```
Comment (stu-gott) [inline docs/live-migration.md]: Whitespace

File: docs/live-migration.md (hunk 14 above)
Comment (admiyo) [inline docs/live-migration.md]: Nice catch.

File: docs/live-migration.md (hunk 12 above)
Comment (admiyo) [inline docs/live-migration.md]: removed.   Can't remember what this was, and seems to not be relevant.

File: docs/live-migration.md [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
+status:
+  phase: Succeeded
+
+```
+
+To cancel `testvm_migration`:
+
+```bash
+
+kubectl delete migrations testvm_migration
```
Comment (stu-gott) [inline docs/live-migration.md]: This line is misleading because the migration has a random suffix which is not acknowledged here.

Comment (rmohr) [inline docs/live-migration.md]: This one does not. Our example migration in cluster/migration.yaml uses `generatwName`.

Comment (admiyo) [inline docs/live-migration.md]: Right.  Should I change this, or update it to use generateName?

**Pull request merged**